lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -1031,6 +862,23 @@ createAsyncOps(scf::ForOp &forOp, tt::CoarseSchedule &schedule,
   // Patch the yield with the updated counters.
   appendToForOpYield(forOp, newYieldOperands);
 
+  // Convert from attributes to CoarseSchedule
+  tt::CoarseSchedule coarseSchedule(numStages);
+  getCoarseSchedule(forOp, coarseSchedule);
+  scheduleDependencies(forOp, coarseSchedule, numStages);
```

Comment:
Instead of calling scheduleDependencies on CoarseSchedule, we can have a different helper scheduleDependenciesWithAttributes(forOp, numStages) to apply (stage, cluster) attributes on the dependent ops.